File: src/main/java/com/history/tree/repositories/TreeRepository.java
package com.history.tree.repositories;

import com.history.tree.model.Tree;
import org.springframework.data.r2dbc.repository.R2dbcRepository;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Mono;

@Repository
public interface TreeRepository extends R2dbcRepository<Tree, Long> {

    Mono<Long> removeById(Long id);

}

File: src/main/kotlin/com/history/tree/repositories/TreeRepository.kt
package com.history.tree.repositories

import com.history.tree.model.Tree
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface TreeRepository : CoroutineCrudRepository<Tree, UUID> {
    suspend fun findByUserId(userId: String)

}